raiden/raiden_service.py

```diff
@@ -412,18 +412,13 @@ def install_and_query_payment_network_filters(
                 from_block,
             )
 
-            chain_id = self.chain.network_id
             # We need to query the events at this point in order to obtain an up to
             # date view of the blockchain right before the node starts its transport
             # and starts receiving messages.
-            for event in self.blockchain_events.poll_blockchain_events(
-                self.get_block_number(),
-            ):
-                on_blockchain_event(
-                    self,
-                    event, event.event_data['block_number'],
-                    chain_id,
-                )
+            chain_id = self.chain.network_id
+            current_block_number = self.chain.client.block_number()
+            for event in self.blockchain_events.poll_blockchain_events(current_block_number):
+                on_blockchain_event(self, event, current_block_number, chain_id)
```

Comment:
Okay, we have a problem here, I only realized it today.

If we do two consecutive restarts, the current_block_number will change, and the behavior of the node **could** change *event though it should not*, and the only solution for this seems to be the block number from the event.

Reply:
Worse than that, I noticed that the above `block_number` call is always returning `None`, which is crashing the node here with `ValueError: {'code': -32602, 'message': 'Invalid params: Invalid variadic value type: invalid type: null, expected a 0x-prefixed, padded, hex-encoded hash with length 64.'}` error on the `toBlock` parameter, maybe because the client may not be initialized yet, both in `master` after the  #1786 was merged, on a clean run, and on top of this branch.  
We need to either fallback to `latest` or let the `_callback_new_block` handle these events =/

Reply:
@andrevmatos I made a follow up PR which removes the need for this value, the events can have a consistent view but user actions and protocol messages cannot, therefor it seems that removing the block number is the best option:

https://github.com/raiden-network/raiden/pull/1859